src/anyp/Uri.cc

```diff
@@ -81,6 +81,48 @@ AnyP::Uri::Encode(const SBuf &buf, const CharacterSet &ignore)
     return output;
 }
 
+SBuf
+AnyP::Uri::Rfc3986Encode(const SBuf &buf)
+{
+    return Encode(buf, CharacterSet::RFC3986_UNRESERVED);
+}
+
+SBuf
+AnyP::Uri::Decode(const SBuf &buf)
+{
+    if (buf.isEmpty())
+        return buf;
+
+    SBuf output;
+    output.reserveSpace(buf.length()); // worst case: buf is not encoded
+
+    const auto end = buf.end();
+    for (auto i = buf.begin(); i != end; ++i) {
+        const auto ch = *i;
+        if (ch != '%') {
+            output.append(ch);
+            continue;
+        }
+        // ch is '%', let's look ahead
+        ++i;
+        if (i == end)
+            throw TextException("incomplete %-encoded triplet at end of input", Here());
+        const auto ch1 = *i;
+        ++i;
+        if (i == end)
+            throw TextException("incomplete %-encoded triplet at end of input", Here());
+        const auto ch2 = *i;
+        const auto hex1 = HextDigitToInt(ch1);
+        const auto hex2 = HextDigitToInt(ch2);
+        if (hex1 == -1 || hex2 == -1)
+            throw TextException("invalid %-encoded triplet", Here());
+
+        const auto decoded = hex1 << 4 | hex2;
```

Comment:
Please make `decoded` a character instead of secretly relying on the current lack of SBuf::append(int) or a similar method or another implicit conversion.

```suggestion
        const auto decoded = static_cast<char>(hex1 << 4 | hex2);
```

Reply:
done, using a type declaration instead of a static_cast